### PR TITLE
Allow to pass more sops arguments

### LINKED ISSUE #46: SOPS collection doesn't see environment variables set with Ansible environment keyword at the task level.
##### SUMMARY

SOPS collection doesn't see environment variables set with Ansible environment keyword at the task level.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

community.sops.load_vars
##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/opt/ansible/lib/python3.6/site-packages/ansible
  executable location = /bin/ansible
  python version = 3.6.8 (default, Nov 16 2020, 16:55:22) [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

CentOS Linux 7 (Core), Linux 3.10.0-1127.10.1.el7.x86_64

##### STEPS TO REPRODUCE

A role with a community.sops.load_vars and environment keyword.

```yaml
- name: Load SOPS Secrets.
  environment:
    AWS_ACCESS_KEY_ID: "{{ AWS_ACCESS_KEY_ID }}"
    AWS_SECRET_ACCESS_KEY: "{{ AWS_SECRET_ACCESS_KEY }}"
  community.sops.load_vars:
    file: "roles/my-role/dev-secrets.yml"
```

##### EXPECTED RESULTS

community.sops.load_vars should use provided environment variables and decrypt secret with KMS key.

##### ACTUAL RESULTS

community.sops.load_vars throws error that access is denied to the key. However same task works if ~/.aws/credentials is present:
[defualt]
AWS_ACCESS_KEY_ID=<>
AWS_SECRET_ACCESS_KEY=<>
So it looks community.sops.load_vars cannot see environment variables set with environment keyword. sops executable works with environment variables: 
AWS_ACCESS_KEY_ID=<> AWS_SECRET_ACCESS_KEY=<> sops --kms "arn:aws:kms:..." -d roles/my-role/dev-secrets.yml

```paste below
ESC[1;30mtask path: /tmp/ansible-pull-common/roles/my-role/tasks/main.yml:41ESC[0m
ESC[0;31mfatal: [test.us-east-1.dev.in]: FAILED! => {"changed": false, "message": "error with file /tmp/ansible-pull-common/roles/my-role/vars/dev-secrets.yml: CouldNotRetrieveKey exited with code 128: Failed to get the data key required to decrypt the SOPS file.\n\nGroup 0: FAILED\n  arn:aws:kms:...: FAILED\n    - | Error decrypting key: AccessDeniedException: The ciphertext\n      | refers to a customer master key that does not exist, does\n      | not exist in this region, or you are not allowed to access.\n      | \tstatus code: 400, request id:\n      | 9c8b4a35-115d-4f64-9685-b86d47e8a66a\n\nRecovery failed because no master key was able to decrypt the file. In\norder for SOPS to recover the file, at least one key has to be successful,\nbut none were.\n"}ESC[0m
```

### PR DESCRIPTION
**Pull request by felixfontein**
### Motivation
Fixes #46.

### Changes description
Adds a set of options to every module and plugin which allows to pass more options that are passed to the sops executable (including an override of the sops binary path).

### Additional notes
The plugin_utils/action_module.py is from https://github.com/ansible-collections/community.crypto/blob/main/plugins/plugin_utils/action_module.py and https://github.com/ansible-collections/community.crypto/blob/main/plugins/module_utils/crypto/module_backends/common.py. I hope that sooner or later this (resp. something similar) is either provided by ansible-base/ansible-core itself, or some common utility collection. For now, I think it's better to vendor the code here instead of making this collection depend on community.crypto just for this piece of utility code.

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

File: tests/integration/targets/lookup_sops/tasks/main.yml [hunk 1]
```diff
@@ -116,3 +116,15 @@
           "a: b\nc: d"
         test_str_binary_data: This is binary data.
         test_str_abcd: a is b, and c is d
+
+    - name: Test fake sops binary
+      set_fact:
+        fake_sops_output: "{{ lookup('community.sops.sops', 'simple.sops.yaml', sops_binary=role_path ~ '/files/fake-sops.sh', enable_local_keyservice=True, aws_access_key_id='xxx') }}"
```
Comment (endorama) [inline tests/integration/targets/lookup_sops/tasks/main.yml:122]: is `~` used as string concatenation here?

Comment (felixfontein) [reply tests/integration/targets/lookup_sops/tasks/main.yml:122]: Yes, that's Jinja2 standard syntax for string concatenation (https://jinja.palletsprojects.com/en/2.11.x/templates/#other-operators).

File: plugins/module_utils/sops.py [hunk 2]
```diff
@@ -41,6 +46,69 @@
...
+    'aws_access_key_id': _create_env_variable('AWS_ACCESS_KEY_ID'),
+    'aws_secret_access_key': _create_env_variable('AWS_SECRET_ACCESS_KEY'),
+    'aws_session_token': _create_env_variable('AWS_SESSION_TOKEN'),
+    'config_path': _create_single_arg('--config'),
+    'enable_local_keyservice': _create_boolean('--enable-local-keyservice'),
+    'keyservice': _create_repeated('--keyservice'),
+}
+
+
+ENCRYPT_OPTIONS = {
```
Comment (endorama) [inline plugins/module_utils/sops.py:97]: I was wondering why dividing the options in 2 different sets. Even if they do apply only to encryption phase, would not be simpler to have them available as in the standard sops cli?

File: plugins/module_utils/sops.py [hunk 3]
```diff
@@ -59,17 +127,39 @@ class Sops():
...
         if input_type is not None:
             command.extend(["--input-type", input_type])
         if output_type is not None:
             command.extend(["--output-type", output_type])
         command.extend(["--decrypt", encrypted_file])
-        process = Popen(command, stdout=PIPE, stderr=PIPE)
-        (output, err) = process.communicate()
-        exit_code = process.returncode
+
+        if module:
```
Comment (endorama) [inline plugins/module_utils/sops.py:157]: May you clarify the usage of `module` here? I think I got the idea. Is something made available by `ActionModuleBase`?

**Comment by endorama** [comment]
I like this addition. It makes everything more complex but I see value in being able to specify options from Ansible code.

@felixfontein as a general question, I think the amount of code may deserve some dedicated unit testing, is something advisable or the current test set is considered enough? (as you manage more collections I expect you have a broader overview than me)

File: plugins/module_utils/sops.py (hunk 2 above)
Comment (felixfontein) [reply plugins/module_utils/sops.py:97]: This reduces the number of options the plugins/modules that do not encrypt have (and thus makes their docs less confusing).

File: plugins/module_utils/sops.py (hunk 3 above)
Comment (felixfontein) [reply plugins/module_utils/sops.py:157]: No, it's for classical `AnsibleModule`s. The Ansible coding guidelines say that modules must never use `subprocess` directly, but use `AnsibleModule.run_command`. This implements that requirement.

**Comment by felixfontein** [comment]
@endorama (more) unit tests would definitely be nice. I've tried to cover most code paths with the existing integration tests, but I didn't manage to cover everything. I think the current test coverage is already pretty good. (Of course it can always be increased :) )

**Comment by felixfontein** [comment]
(Rebased against `main`.)

**Pull request merged**